### ISSUE TITLE
The WebSockets Transport is not closing the connections graciously

### ISSUE THREAD

**Issue by diasdavid** [libp2p/js-libp2p#139]
Found while developing the PDD Stories https://github.com/libp2p/js-libp2p/pull/137.

To repro:

```
git@github.com:libp2p/js-libp2p.git -b pdd
cd js-libp2p
npm install
cd pdd
npm install
# now in two terminal windows
# terminal 1
node pdd-transport--story-2--peer-b.js
# terminal 2
node pdd-transport--story-2--peer-a.js
```

The expected result should be that the two process ends successfully, however, peer-b never ends as it never received the end event.

**Comment by mkg20001** [comment]
Actually it's a bug in the test:
https://github.com/libp2p/js-libp2p/blob/master/pdd/pdd-transport--story-2--peer-b.js#L36-L46
This connects the sink and source of conn and returns the `pull.onEnd` because `onEnd` is a sink and not a through. See: https://github.com/pull-stream/pull-stream/blob/master/sinks/on-end.js

I've created a test for that case but haven't committed that yet. As this does not seem a bug in the module, should I still commit that test?

**Comment by mkg20001** [comment]
Added fix to pr #150
